### PR TITLE
docs(demo): add example of ngx-datatable integration

### PR DESCRIPTION
**Pull request by enricooliva**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
add docs with example of datatable integration 
ngx-datatable

issue: custom type with ngx-datatable #1061

**Please check if the PR fulfills these requirements**
- [ ] The commit messages follow our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] A unit test has been written for this change.
- [ ] Running `npm run build` produced a successful build. (Unit testing can be done by running `npm test`;)
- [ ] My code has been linted. (`npm run lint` to do this. `npm run build` will fail if there are files not linted.)

**Please provide a screenshot of this feature before and after your code changes, if applicable.**
ex

**Other information**:

@aitboudad  I added the example.

### CODE REVIEW

File: demo/src/app/examples/advanced/datatable-integration/datatable.type.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  [scrollbarH]="scrollbarH"      
+  [reorderable]="reorderable"    
+  [externalSorting]="true"
+  [selected]="selected"
+  [selectionType]="'single'"
+  (sort)="onSort($event)"
+  (select)='onSelect($event)'>     
+  
+  <ng-template #genericcolumn ngx-datatable-cell-template let-rowIndex="rowIndex" let-value="value" let-row="row" let-column="column">
+    <formly-field
```
Comment (aitboudad) [inline demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: use `formly-group` instead

File: demo/src/app/examples/advanced/datatable-integration/datatable.type.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+})
+
+export class DatatableTypeComponent extends FieldArrayType {
+  constructor(builder: FormlyFormBuilder) {
+    super(builder);
+  }
+
+  @ViewChild('genericcolumn') public genericcolumn: TemplateRef<any>;
+  
+  @Input() columnMode: any = "force";
```
Comment (aitboudad) [inline demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: I would prefer to move these options into `defaultOptions` declaration, see the datepicker example [datepicker.module.ts#L26-L32](https://github.com/formly-js/ngx-formly/blob/master/src/material/datepicker/src/datepicker.module.ts#L26-L32)

with that way those option can be passed through `fields`

File: demo/src/app/examples/advanced/datatable-integration/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+@NgModule({
+  imports: [
+    CommonModule,
+    ReactiveFormsModule,
+    FormlyBootstrapModule,
+    NgxDatatableModule,
+    FormlyModule.forRoot({
+      types: [
+        { name: 'table', component: DatatableTypeComponent },
```
Comment (aitboudad) [inline demo/src/app/examples/advanced/datatable-integration/app.module.ts]: I would use `datatable` for name

File: demo/src/app/examples/advanced/datatable-integration/datatable.type.ts (hunk 1 above)
Comment (enricooliva) [reply demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: I changed `getFields` in `getField` and continue to use `formly-field`

File: demo/src/app/examples/advanced/datatable-integration/datatable.type.ts (hunk 2 above)
Comment (enricooliva) [reply demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: ok thanks

File: demo/src/app/examples/advanced/datatable-integration/app.module.ts (hunk 3 above)
Comment (enricooliva) [reply demo/src/app/examples/advanced/datatable-integration/app.module.ts]: ok

File: demo/src/app/examples/advanced/datatable-integration/datatable.type.ts (hunk 1 above)
Comment (enricooliva) [reply demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: The ng-template is a template for each column and used for each cell. Each cell match with the single field in this sense I think is correct to use formly-field or no?

Comment (enricooliva) [reply demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: @aitboudad you have any suggestion?

Comment (aitboudad) [reply demo/src/app/examples/advanced/datatable-integration/datatable.type.ts]: it's ok then!

**Comment by aitboudad** [comment]
Thank you @enricooliva!

**Pull request merged**